pkg/scheduler/framework/detector/detect_framework_test.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This file helps to detect whether we changed the interface of Framework.
+// This is important for downstream projects who import the Framework to not
+// breaking their codes.
+
+package detector
+
+import (
+	"context"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/kubernetes/pkg/scheduler/apis/config"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+)
+
+var _ framework.Framework = &detect_framework{}
+
+type detect_framework struct {
+	framework.Handle
+}
+
+var (
+	fakeStatus          = framework.NewStatus(framework.Success)
+	fakePreFilterResult = framework.PreFilterResult{
+		NodeNames: sets.NewString("fake-node"),
+	}
+	fakePostFilterResult = framework.PostFilterResult{
+		NominatingInfo: &framework.NominatingInfo{
+			NominatedNodeName: "fake-node",
+			NominatingMode:    framework.ModeNoop,
+		},
+	}
+	defaultSchedulerName = "default-scheduler"
+)
+
+func (d *detect_framework) QueueSortFunc() framework.LessFunc {
+	return func(_, _ *framework.QueuedPodInfo) bool { return false }
+}
+
+func (d *detect_framework) RunPreFilterPlugins(ctx context.Context, state *framework.CycleState, pod *v1.Pod) (*framework.PreFilterResult, *framework.Status) {
+	return &fakePreFilterResult, fakeStatus
+}
+
+func (d *detect_framework) RunPostFilterPlugins(ctx context.Context, state *framework.CycleState, pod *v1.Pod, filteredNodeStatusMap framework.NodeToStatusMap) (*framework.PostFilterResult, *framework.Status) {
+	return &fakePostFilterResult, fakeStatus
+}
+
+func (d *detect_framework) RunPreBindPlugins(ctx context.Context, state *framework.CycleState, pod *v1.Pod, nodeName string) *framework.Status {
```

Comment:
Does cluster autoscaler implement this?

Reply:
I checked the project, they didn't. cc @x13n for confirmation.

Reply:
Good point, we should only be using `RunPreFilterPlugins` and `RunFilterPlugins` in Cluster Autoscaler (and `NewFramework` & `NewCycleState` functions). Checking for the entire interface might be unnecessarily constraining. Maybe just define subset of the interface and check the conversion? Something like this:
- [happy case](https://go.dev/play/p/53BLWq67uJL)
- [failing case](https://go.dev/play/p/1Z6RsCfDgxl)

Reply:
updated